src/Arcus.EventGrid/Parsers/EventGridParser.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Arcus.EventGrid.Contracts.Interfaces;
+using GuardNet;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Arcus.EventGrid.Parsers
+{
+    public class EventGridParser
+    {
+        /// <summary>
+        ///     Parses a string to a EventGridMessage with typed data payload
+        /// </summary>
+        /// <param name="rawJsonBody">Raw JSON body</param>
+        /// <returns>Typed EventGridMessage</returns>
+        public static EventGridMessage<TEvent> Parse<TEvent>(string rawJsonBody)
+            where TEvent : IEvent, new()
+        {
+            Guard.NotNullOrWhitespace(rawJsonBody, nameof(rawJsonBody));
+            var sessionId = Guid.NewGuid().ToString();
+
+            var eventGridMessage = Parse<TEvent>(rawJsonBody, sessionId);
+            return eventGridMessage;
+        }
+
+        /// <summary>
+        ///     Parses a string to a EventGridMessage with typed data payload
+        /// </summary>
+        /// <param name="rawJsonBody">Raw JSON body</param>
+        /// <param name="sessionId">Session id for event grid message</param>
+        /// <returns>Typed EventGridMessage</returns>
+        public static EventGridMessage<TEvent> Parse<TEvent>(string rawJsonBody, string sessionId)
```

Comment:
These parameters should also be guarded?

Reply:
Good point, adding it along with tests